redaxo/src/addons/structure/lib/linkmap/var_linklist.php

```diff
@@ -56,11 +56,13 @@ public static function getWidget($id, $name, $value, array $args = [])
         $options = '';
         $linklistarray = explode(',', $value);
         foreach ($linklistarray as $link) {
-            if ('' != $link) {
-                if ($article = rex_article::get((int) $link)) {
-                    $options .= '<option value="' . $link . '">' . rex_escape(trim(sprintf('%s [%s]', $article->getName(), $article->getId()))) . '</option>';
-                }
+            if ('' == $link) {
+                continue;
+            }
+            if (!($article = rex_article::get((int) $link))) {
+                continue;
             }
+            $options .= '<option value="' . $link . '">' . rex_escape(trim(sprintf('%s [%s]', $article->getName(), $article->getId()))) . '</option>';
```

Comment:
Hier hätte ich beim selbst schreiben das obere if auch rausgetrennt, aber beim unteren hätte ich den Code im if gelassen.
Weil das if inline die Variable `$article` erstellt, quasi für den Scope des if-blocks (auch wenn es in php technisch den Scope so nicht gibt).

(Aber kann auch hier mit der generierten Variante leben)

Reply:
kann man dem rector beibringen wenn es uns das wert ist ;)

Reply:
dazu müsste der rector aber auch ermitteln, dass die variable nur im if scope genutzt wird und nirgends anders im aktuellen scope